src/module/hotKeysModule.js

```diff
@@ -225,6 +225,42 @@ var HotKeysModule = class HotKeysModule {
             }
         );
 
+        [...Array(10).keys()].forEach((workspaceIndex) => {
+            const actionKey = `MOVE_WINDOW_TO_${workspaceIndex + 1}`;
+            KeyBindingAction[actionKey] = `move-window-to-workspace-${
+                workspaceIndex + 1
+            }`;
+
+            this.actionNameToActionMap.set(KeyBindingAction[actionKey], () => {
+                const activeMsWorkspace = Me.msWorkspaceManager.getActivePrimaryMsWorkspace();
+                if (
+                    activeMsWorkspace.tileableFocused ===
+                    activeMsWorkspace.appLauncher
+                ) {
+                    return;
+                }
+
+                if (
+                    activeMsWorkspace ===
+                    Me.msWorkspaceManager.primaryMsWorkspaces[
+                        Me.msWorkspaceManager.primaryMsWorkspaces.length - 2
+                    ]
+                ) {
+                    return;
+                }
+
+                const nextMsWorkspace =
+                    Me.msWorkspaceManager.primaryMsWorkspaces[
+                        workspaceIndex
+                    ];
```

Comment:
I think it should test if the key combo refers to the current works (and abort in this case) and also if ir refers to a number greater than the actually available workspaces (and then move to the last available in this case, maybe?).